compiler/optimizer/VPHandlers.cpp

```diff
@@ -5704,6 +5704,16 @@ TR::Node *constrainAcall(TR::ValuePropagation *vp, TR::Node *node)
             TR::VPResolvedClass *newTypeConstraint = NULL;
             if (constraint)
                {
+               // Do nothing if the class of the object doesn't implement Cloneable
+               if (constraint->getClass() && !vp->comp()->fej9()->isCloneable(constraint->getClass()))
```

Comment:
I don't think the site would have been queued for transformation, but I agree this could be made extra safe here with a ! and an ||.